thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/pinot/PinotThirdEyeClientConfig.java

```diff
@@ -94,12 +81,31 @@ public String toString() {
     return stringHelper.toString();
   }
 
-
-  public static PinotThirdEyeClientConfig createThirdEyeClientConfig(ThirdEyeConfiguration config) throws Exception {
-    File clientConfigDir = new File(config.getRootDir(), "client-config");
-    File clientConfigFile = new File(clientConfigDir, config.getClient() + ".yml");
-    PinotThirdEyeClientConfig thirdEyeClientConfig =
-        PinotThirdEyeClientConfig.fromFile(clientConfigFile);
-    return thirdEyeClientConfig;
+  public static PinotThirdEyeClientConfig createThirdeyeClientConfig(ClientConfig clientConfig) {
+    PinotThirdEyeClientConfig config = new PinotThirdEyeClientConfig();
+    for (Client client : clientConfig.getClients()) {
+      if (client.getClassName().equals(PinotThirdEyeClient.class.getCanonicalName())) {
+        Map<String, String> properties = client.getProperties();
+        String controllerHost = properties.get(PinotThirdeyeClientProperties.CONTROLLER_HOST.getValue());
+        int controllerPort = Integer.valueOf(properties.get(PinotThirdeyeClientProperties.CONTROLLER_PORT.getValue()));
+        String clusterName = properties.get(PinotThirdeyeClientProperties.CLUSTER_NAME.getValue());
+        String zookeeperUrl = properties.get(PinotThirdeyeClientProperties.ZOOKEEPER_URL.getValue());
+        String brokerUrl = properties.get(PinotThirdeyeClientProperties.BROKER_URL.getValue());
+        String tag = properties.get(PinotThirdeyeClientProperties.TAG.getValue());
+
+        config.setControllerHost(controllerHost);
+        config.setControllerPort(controllerPort);
+        config.setClusterName(clusterName);
+        config.setZookeeperUrl(zookeeperUrl);
+        if (StringUtils.isNotBlank(brokerUrl)) {
+          config.setBrokerUrl(brokerUrl);
+        }
+        if (StringUtils.isNotBlank(tag)) {
+          config.setTag(tag);
+        }
+        break;
+      }
+    }
+    return config;
```

Comment:
I recommend adding at least some javadoc stating that there can only be a single instance of the Pinot client. Or better yet, throw an exception if there's multiple.